### PR TITLE
chore: Add to workflows

### PR DESCRIPTION
**Pull request by ebebbington**
## Summary

* Add pre release workflow
* Add social jobs

### CODE REVIEW

**Comment by ebebbington** [comment]
Ok deffo ready for review

File: .github/workflows/pre_release.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          curl -XPOST -u "${{ secrets.CI_USER_NAME }}:${{ secrets.CI_USER_PAT }}" -H "Accept: application/vnd.github.everest-preview+json" -H "Content-Type: application/json" https://api.github.com/repos/drashland/website/dispatches --data '{"event_type": "update_docs", "client_payload": { "module_to_update": "sockets", "version": "${{ github.ref }}" }}'
+
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v2
+        with:
+          token: ${{ secrets.CI_USER_PAT }}
+          commit-message: bump versions
+          title: bump versions
+          body: This was auto-generated by GitHub Actions.
+          branch: release # Can change it to ${{ github.ref }} to be the branch name, but then we need to figure out how to stop this worjflow running  when it creates another "release-vx.x.x" branch
```
Comment (crookse) [inline .github/workflows/pre_release.yml:49]: typo: worjflow

File: console/bumper_ci_service_files.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    replaceWith: `"version": "{{ thisModulesLatestVersion }}"`,
+  },
+  {
+    filename: "./README.md",
+    replaceTheRegex: regexes.import_export_statements,
+    replaceWith: `sockets@v{{ thisModulesLatestVersion }}`,
+  },
+];
+
+export const denoVersionFiles = [
```
Comment (crookse) [inline console/bumper_ci_service_files.ts:21]: `bumperFiles`

File: console/bumper_ci_service_files.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+export const regexes = {
+  const_statements: /version = ".+"/g,
+  egg_json: /"version": ".+"/,
+  import_export_statements: /sockets@v[0-9\.]+[0-9\.]+[0-9\.]/g,
+  yml_deno: /deno: \[".+"\]/g,
+};
+
+export const moduleVersionFiles = [
```
Comment (crookse) [inline console/bumper_ci_service_files.ts:8]: `preReleaseFiles`

File: console/bumper_ci_service.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { BumperService } from "https://raw.githubusercontent.com/drashland/services/master/ci/bumper_service.ts";
+import {
+  denoVersionFiles,
+  moduleVersionFiles,
+} from "./bumper_ci_service_files.ts";
+
+const b = new BumperService("sockets", Deno.args);
+
+if (b.isForPreRelease()) {
+  b.bump(moduleVersionFiles);
```
Comment (crookse) [inline console/bumper_ci_service.ts:10]: `preReleaseFiles`

File: console/bumper_ci_service.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+} else {
+  b.bump(denoVersionFiles);
```
Comment (crookse) [inline console/bumper_ci_service.ts:12]: `bumperFiles`

Comment (ebebbington) [reply console/bumper_ci_service.ts]: https://github.com/drashland/sockets/pull/95/commits/2fbec0d6f7cf8cdfcc7b082d6445dac69a796752

File: console/bumper_ci_service.ts (hunk 4 above)
Comment (ebebbington) [reply console/bumper_ci_service.ts]: https://github.com/drashland/sockets/pull/95/commits/2fbec0d6f7cf8cdfcc7b082d6445dac69a796752

File: console/bumper_ci_service_files.ts (hunk 3 above)
Comment (ebebbington) [reply console/bumper_ci_service_files.ts]: https://github.com/drashland/sockets/pull/95/commits/2fbec0d6f7cf8cdfcc7b082d6445dac69a796752

File: console/bumper_ci_service_files.ts (hunk 2 above)
Comment (ebebbington) [reply console/bumper_ci_service_files.ts]: https://github.com/drashland/sockets/pull/95/commits/2fbec0d6f7cf8cdfcc7b082d6445dac69a796752

File: .github/workflows/pre_release.yml (hunk 1 above)
Comment (ebebbington) [reply .github/workflows/pre_release.yml]: https://github.com/drashland/sockets/pull/95/commits/2fbec0d6f7cf8cdfcc7b082d6445dac69a796752

File: .github/workflows/pre_release.yml [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Checkout master
+        uses: actions/checkout@v2
+        with:
+          ref: master
+
+      - name: Delete Remote Release Branch
```
Comment (crookse) [inline .github/workflows/pre_release.yml:27]: comment out?

Comment (ebebbington) [reply .github/workflows/pre_release.yml:27]: good catch, i swear down i commented that out :S

Comment (ebebbington) [reply .github/workflows/pre_release.yml]: https://github.com/drashland/sockets/pull/95/commits/e257cb80032e672c15e1164888981cacc39cf81e

**Review by crookse — APPROVED** [review]
image

**Pull request merged**